src/sumlog.jl

```diff
@@ -0,0 +1,36 @@
+using IrrationalConstants: logtwo
+
+"""
+$(SIGNATURES)
+
+Compute `sum(log.(X))`.
+
+`sum(log.(X))` can be evaluated much more quickly as `sum(log, X)`. However,
+this still requires computing `log` for each element of `X`. 
+
+`sumlog(X)` can be faster still, especially an the length of `X` increases.
```

Comment:
Maybe this could be shortened a bit to make it immediately clear what's the advantage of this function:
```suggestion
Compute `sum(log.(X))` with a single `log` evaluation.

This is faster than computing `sum(log.(X))` or even `sum(log, X)`, in particular as `X` increases.
```